llvm/include/llvm/CAS/ObjectStore.h

```diff
@@ -376,6 +376,13 @@ using ObjectStoreCreateFuncTy =
     Expected<std::unique_ptr<ObjectStore>>(const Twine &);
 void registerCASURLScheme(StringRef Prefix, ObjectStoreCreateFuncTy *Func);
 
+class ActionCache;
+
+/// Create on-disk \p ObjectStore and \p ActionCache instances based on \p
```

Comment:
Why don't just put this in UnifiedOnDiskCache.h?

Reply:
I want to keep `UnifiedOnDiskCache` (header and C++ file) free of dependencies to `ObjectStore` or `ActionCache`.
I intend to create a Swift package with a Swift wrapper interface over `UnifiedOnDiskCache`; it will get copies of llvm sources but I'd only need a subset of the files from `lib/CAS`.

Reply:
I think we either let this function return `UnifiedOnDiskCache` and switch away from using ObjectStore and ActionCache in clang, or maybe just rename this function for now to something that doesn't get associated with `UnifiedOnDiskCache`

Reply:
> I think we either let this function return UnifiedOnDiskCache and switch away from using ObjectStore and ActionCache in clang

clang needs to be using ObjectStore/ActionCache APIs since there's going to be another implementation of these for clang to use, via the plugin.

> or maybe just rename this function for now to something that doesn't get associated with UnifiedOnDiskCache

My concern with that is that it will become unclear what the difference is between this function and `createOnDiskCAS`.
I'll pull `createOnDiskUnifiedCASDatabases` out of `ObjectStore.h` and put it in a separate header.

Reply:
I moved `createOnDiskUnifiedCASDatabases` out of `ObjectStore` here: https://github.com/apple/llvm-project/pull/6312/commits/476b6444343d7c01ace21bd92aa5c723dd3864d7